Enterprise-CMCS/macpro-mdct-mfp
# [CMDCT-3032] Fix nested choicelist not saving not-last children answers when tabbing off

### Description


This is bringing over a patch from MCR that I made to fix nested textfields and choices from acting funny. They'd show the wrong thing, or clear other fields out if the field wasn't the last thing in the choicelist. Heres a before:

**Before - Original Bug**
https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/22242b2c-0638-4c47-914d-80ca02a3dfda

**After**
https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/24170298-a79c-4115-b6f8-62006e6eb5a2

### Related ticket(s)


CMDCT-3032

---
### How to test

Create a Work Plan
You'll want to go to getEntityStatus.tsx and comment out this line on 120: `if (stepType === "closeOutInformation") return "disabled";`. That way you can get to the closeout step
Create an Initiative
Go to Close out
Play around with the nested choices and see if it breaks

---
### Author checklist
